### DISCUSSION TITLE
Issues While Doing Undo

### DISCUSSION THREAD

**Discussion by RakulAgn** [GrapesJS/grapesjs#5151] [Q&A]
While Fetching template from Api and loaded to editor if we give undo the whole template gets cleared actually this is the way it will happen or any alternative solution is there ? I am expecting once if there is changes happened after the template load on this point of view the undo should be worked..

**Comment by artf** [comment]
Sorry, can you elaborate better what is the issue and the expectations?

**Comment by asimex** [comment]
editor.setComponents(importedData);
            editor.UndoManager.clear();

check this it worked for me

**Reply by glaprida** [reply]
As far as I understand, `clear()` erases the undo stack, so you won't be able to undo whatever changes you made prior to the API call.

I haven't tried this, but if I'm interpreting the Docs right, you should wrap your API call inside 
```
editor.undoManager.skip(() => {
 // Do stuff without tracking -- like the setComponents(importedData);
});
```
